lib/ClangImporter/ImportType.cpp

```diff
@@ -510,36 +510,32 @@ namespace {
     }
 
     ImportResult VisitVectorType(const clang::VectorType *type) {
-      auto *SIMD = Impl.tryLoadSIMDModule();
-      if (!SIMD)
-        return Type();
-      
-      // Map the element type and count to a Swift name, such as
-      // float x 3 => Float3.
-      SmallString<16> name;
-      {
-        llvm::raw_svector_ostream names(name);
-        
-        if (auto builtinTy
-              = dyn_cast<clang::BuiltinType>(type->getElementType())){
-          switch (builtinTy->getKind()) {
-#define MAP_SIMD_TYPE(TYPE_NAME, __, BUILTIN_KIND)   \
-          case clang::BuiltinType::BUILTIN_KIND: \
-            names << #TYPE_NAME;                 \
-            break;
-#include "swift/ClangImporter/SIMDMappedTypes.def"
-          default:
-            // A vector type we don't know how to map.
-            return Type();
+      // Get the element type and count.
+      Type element = Impl.importTypeIgnoreIUO(type->getElementType(),
+                                              ImportTypeKind::Abstract,
+                                              false,
+                                              Bridgeability::Full,
+                                              OptionalTypeKind::OTK_None);
+      unsigned count = type->getNumElements();
+      // Import vector-of-one as the element type.
+      if (count == 1) { return element; }
```

Comment:
Hmm.  Does this always have the same layout as it would in C?

Reply:
Up to manually-specified alignment, I think so. (How do we handle int32 typedefs that are over/under-aligned for that matter?)

Reply:
We probably ignore the alignment, which is probably a bug. :)

Reply:
As long as it's consistently a bug, the same as other imported types =)